diff --git a/nss/lib/sysinit/nsssysinit.c b/nss/lib/sysinit/nsssysinit.c
index bd0fac2..73650cb 100644
--- a/nss/lib/sysinit/nsssysinit.c
+++ b/nss/lib/sysinit/nsssysinit.c
@@ -54,11 +54,11 @@ appendDirAndCreate(char *path, char *dir, mode_t mode)
     return 0;
 }
 
-#define XDG_NSS_USER_PATH1 "/.local"
-#define XDG_NSS_USER_PATH2 "/share"
+#define XDG_NSS_USER_PATH1 "/var"
+#define XDG_NSS_USER_PATH2 "/lib"
 #define XDG_NSS_USER_PATH3 "/pki"
 
-#define NSS_USER_PATH1 "/.pki"
+#define NSS_USER_PATH1 "/var/cache/pki"
 #define NSS_USER_PATH2 "/nssdb"
 
 /**
