diff --git a/config.def.h b/config.def.h
index b5b26bf..f0a05e3 100644
--- a/config.def.h
+++ b/config.def.h
@@ -7,11 +7,6 @@
  * KEY_CODE_YES).
  */
 
-/* Old versions of ncurses don't support A_ITALIC.
- * Define this to disable it if the situation isn't automatically detected.
-#define NO_ITALICS
- */
-
 /* How often to check for status updates by default. */
 #define STATUSINTERVAL 60
 
@@ -36,16 +31,6 @@
 /* The force redraw key. */
 #define REDRAW KEY(L'l')
 
-/* The path for the wide-character curses library. */
-#ifndef NCURSESW_INCLUDE_H
-    #if defined(__APPLE__) || (defined(BSD) && !defined(__linux__))
-        #define NCURSESW_INCLUDE_H <curses.h>
-    #else
-        #define NCURSESW_INCLUDE_H <ncursesw/curses.h>
-    #endif
-#endif
-#include NCURSESW_INCLUDE_H
-
 /* Includes needed to make forkpty(3) work. */
 #ifndef FORKPTY_INCLUDE_H
     #if defined(__APPLE__) || (defined(BSD) && !defined(__linux__))
diff --git a/mtm.c b/mtm.c
index 14cc04d..e40d0de 100644
--- a/mtm.c
+++ b/mtm.c
@@ -17,6 +17,7 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <locale.h>
+#include <ncurses.h>
 #include <pwd.h>
 #include <signal.h>
 #include <stdbool.h>
