diff --git a/src/os_unix.h b/src/os_unix.h
index 5df943c..80bdf16 100644
--- a/src/os_unix.h
+++ b/src/os_unix.h
@@ -248,7 +248,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define USR_EXRC_FILE "sys$login:.exrc"
 # else
-#  define USR_EXRC_FILE "$HOME/.exrc"
+#  define USR_EXRC_FILE "$HOME/etc/prog.d/exrc"
 # endif
 #endif
 
@@ -260,7 +260,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 # define USR_VIMRC_FILE  "sys$login:.vimrc"
 # else
-#  define USR_VIMRC_FILE "$HOME/.vimrc"
+#  define USR_VIMRC_FILE "$HOME/etc/vimrc"
 # endif
 #endif
 
@@ -269,7 +269,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define USR_VIMRC_FILE2	"sys$login:vimfiles/vimrc"
 # else
-#   define USR_VIMRC_FILE2	"~/.vim/vimrc"
+#   define USR_VIMRC_FILE2	"~/etc/vim.d/vimrc"
 # endif
 #endif
 
@@ -312,7 +312,7 @@ typedef struct dsc$descriptor   DESC;
 #  ifdef VMS
 #   define VIMINFO_FILE  "sys$login:.viminfo"
 #  else
-#   define VIMINFO_FILE "$HOME/.viminfo"
+#   define VIMINFO_FILE "$HOME/var/cache/viminfo"
 #  endif
 # endif
 # if !defined(VIMINFO_FILE2) && defined(VMS)
@@ -325,7 +325,7 @@ typedef struct dsc$descriptor   DESC;
 #endif
 
 #ifndef VIMRC_FILE
-# define VIMRC_FILE	".vimrc"
+# define VIMRC_FILE	"vimrc"
 #endif
 
 #ifdef FEAT_GUI
@@ -342,7 +342,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define DFLT_BDIR    "./,sys$login:,tmp:"
 # else
-#  define DFLT_BDIR    ".,~/tmp,~/"    /* default for 'backupdir' */
+#  define DFLT_BDIR    "~/tmp"    /* default for 'backupdir' */
 # endif
 #endif
 
@@ -350,7 +350,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define DFLT_DIR     "./,sys$login:,tmp:"
 # else
-#  define DFLT_DIR     ".,~/tmp,/var/tmp,/tmp" /* default for 'directory' */
+#  define DFLT_DIR     "~/tmp" /* default for 'directory' */
 # endif
 #endif
 
@@ -358,7 +358,7 @@ typedef struct dsc$descriptor   DESC;
 # ifdef VMS
 #  define DFLT_VDIR    "sys$login:vimfiles/view"
 # else
-#  define DFLT_VDIR    "$HOME/.vim/view"       /* default for 'viewdir' */
+#  define DFLT_VDIR    "$HOME/etc/vim.d/view"       /* default for 'viewdir' */
 # endif
 #endif
 
@@ -370,15 +370,15 @@ typedef struct dsc$descriptor   DESC;
 #else
 # ifdef RUNTIME_GLOBAL
 #  ifdef RUNTIME_GLOBAL_AFTER
-#   define DFLT_RUNTIMEPATH	"~/.vim," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL_AFTER ",~/.vim/after"
-#   define CLEAN_RUNTIMEPATH	RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL_AFTER
+#   define DFLT_RUNTIMEPATH	"~/etc/vim.d"
+#   define CLEAN_RUNTIMEPATH	"~/etc/vim.d"
 #  else
-#   define DFLT_RUNTIMEPATH	"~/.vim," RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL "/after,~/.vim/after"
-#   define CLEAN_RUNTIMEPATH	RUNTIME_GLOBAL ",$VIMRUNTIME," RUNTIME_GLOBAL "/after"
+#   define DFLT_RUNTIMEPATH	"~/etc/vim.d"
+#   define CLEAN_RUNTIMEPATH	"~/etc/vim.d"
 #  endif
 # else
-#  define DFLT_RUNTIMEPATH	"~/.vim,$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after,~/.vim/after"
-#  define CLEAN_RUNTIMEPATH	"$VIM/vimfiles,$VIMRUNTIME,$VIM/vimfiles/after"
+#  define DFLT_RUNTIMEPATH	"~/etc/vim.d"
+#  define CLEAN_RUNTIMEPATH	"~/etc/vim.d"
 # endif
 #endif
 
@@ -390,7 +390,7 @@ typedef struct dsc$descriptor   DESC;
 # define TEMPNAMELEN    28
 #else
 /* Try several directories to put the temp files. */
-# define TEMPDIRNAMES  "$TMPDIR", "/tmp", ".", "$HOME"
+# define TEMPDIRNAMES   "~/tmp"
 # define TEMPNAMELEN    256
 #endif
 
